### ISSUE TITLE
Cannot open include file curand.h. No such file or directory

### ISSUE THREAD

**Issue by git-sohib** [AlexeyAB/darknet#951]
For some reason I had to reinstall windows and other software. Now, I am trying to run  darknet.sln. But it  is not working. Before reinstallation I used to run darknet.sln without any issue with CUDA 8.0 and corresponding vwersion of cudnn. Now, I would like to move a step up and configure it with Cuda 9.1. I did all the steps mentioned in **How to compile on Windows:** Now I am building it and then I have errors that are mainly saying that _Cannot open include file curand.h. No such file or directory_. I found out that I had no that file in my C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.1\include path and copied it from other PC that has **curand.h.** file. If I run now I have more errors?
 
HOW SHOULD FIX THIS?

**Comment by AlexeyAB** [comment]
Just re-install CUDA 9.1: https://developer.nvidia.com/cuda-91-download-archive

curand.h should be in the `C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.1\include`

**Comment by git-sohib** [comment]
@AlexeyAB Thank you, why would that error occur? I remember I checked for checksums of **cuda_9.1.85_win10_network**.

**Comment by AlexeyAB** [comment]
Incorrect CUDA installation.
May be you have several CUDA versions and some of them are broken.

**Comment by git-sohib** [comment]
@AlexeyAB I only installed Base Installer and 3 Patch releases of referenced link. Anyway, let me try and if there is something I'll be back again.

Кстати, большое спасибо! ))

**Comment by git-sohib** [comment]
@AlexeyAB 
I've set everything as is. But now I have other errors [here:](https://habrastorage.org/webt/1r/gb/ag/1rgbagngdyjkcrh9edtxdfoeogc.png)

**Comment by git-sohib** [comment]
Greetings Mr. @AlexeyAB 
- I built `darknet_no_gpu.sln` succesfully
-Yes I reinstalled it
- Yes, I use MSVS 2015
- No I didn't change anything after my windows reinstallation.As I said before that it was working properly.
-Sure, here it is

[](https://habrastorage.org/webt/ot/qg/j-/otqgj-wvqcdo4wotxuhkn6pivdy.png)

**Comment by git-sohib** [comment]
@AlexeyAB 
![](https://habrastorage.org/webt/8w/qm/lv/8wqmlvegyejbrwuoj5sfcckok7u.png)

![](https://habrastorage.org/webt/q3/jm/wm/q3jmwmevk7wa8ieyzao4q96lddw.png)

**Comment by AlexeyAB** [comment]
> Now I am building it and then I have errors that are mainly saying that Cannot open include file curand.h. No such file or directory. I found out that I had no that file in my C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.1\include path and copied it from another PC that has curand.h. file. If I run now I have more errors

1. As I see curand.h is here. Did you copy it from another places manually? If yes - then remove your CUDA and install it again.

2. Did you compile x64 & Release?

3. Do you use default `darknet.sln` without changes?

4. Can you show Full-screen screenshot of MSVS2015 when errors occurred?

**Comment by git-sohib** [comment]
@AlexeyAB 
- I copied it from another PC first, but then you said re-install CUDA 9.1. Before re-installing I deleted all CUDA folders, un-installed CUDA, and re-installed CUDA 9.1 with its 3 patch releases and installed in order. So there must not have been old files anyway.
- Yes, I compiled it with x64 & Realise
- Yes, I am using darknet.sln without any change since my last successful run
- for this please refer to [here](https://habrastorage.org/webt/1r/gb/ag/1rgbagngdyjkcrh9edtxdfoeogc.png)

**Comment by AlexeyAB** [comment]
@git-sohib 

>  I am using darknet.sln without any change since my last successful run

Try to delete Darknet and download it again. Then use `darknet.sln` without any changes.

**Comment by git-sohib** [comment]
yes I did Mr. @AlexeyAB. It helped but I installed another forked and customized version of darknet. Thank you for your time!
